cpp/include/raft/sparse/neighbors/detail/connect_components.cuh

```diff
@@ -43,16 +47,21 @@
 #include <thrust/transform.h>
 #include <thrust/tuple.h>
 
+#include <thrust/gather.h>
+#include <thrust/scatter.h>
+
 #include <cub/cub.cuh>
 
 #include <limits>
 
 namespace raft::sparse::neighbors::detail {
 
 /**
- * Functor with reduction ops for performing fused 1-nn
- * computation and guaranteeing only cross-component
- * neighbors are considered.
+ * Functor with reduction ops for performing masked 1-nn
+ * computation. this change introduces a breaking change to
```

Comment:
I had re-introduced the colors array here (forgot to re-update the docs), since having it will not modify the set of cross-component neighbors (though it is slightly suboptimal because we don't really need the colors) and we can get away with not having to modify the public API of the reductionOp.

Reply:
So now I deleted this note about `colors`.